### PR TITLE
OASIS Brain DCGAN

### PR DESCRIPTION
**Pull request by ldgrp**
Add Oasis Brain DCGAN implementation

Leo Orpilla
Student #: 45295517

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Work (mostly) submitted before deadline
- All commits are made in 2 days
- Insufficient details in commit messages 
- Code appear valid
- Results are good 
Other Feedback
Good code structure/modularisation. Now sure how you did distributed computing since there is only 1 gpu per lab PC

**Comment by rexsw** [comment]
Code looks good for hard difficulty

For next time;

Tensorflow has functions for image opening and more so there’s no need to also use PIL in your project. Same with numpy and shuffling the data 

Comments such as 
   def epoch_step(self, images):
        '''A step in the epoch.'''
Add nothing to the readability of the code and should be avoided. Overall comments were well done

File: recognition/ldgrp-OasisDCGAN/driver.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+from typing import Optional
```
Comment (shakes76) [inline recognition/ldgrp-OasisDCGAN/driver.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Proper Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Pull request merged**